lib/make-client.ts

```diff
@@ -1,26 +1,32 @@
 import createContentfulApi, {
-  ClientWithAllLocalesAndWithLinkResolution,
+  ClientWithAllLocalesAndWithLinkResolutionAndWithoutUnresolvableLinks,
+  ClientWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks,
   ClientWithAllLocalesAndWithoutLinkResolution,
-  ClientWithLinkResolution,
+  ClientWithLinkResolutionAndWithoutUnresolvableLinks,
+  ClientWithLinkResolutionAndWithUnresolvableLinks,
   ClientWithoutLinkResolution,
   ContentfulClientApi,
   CreateContentfulApiParams,
   DefaultClient,
 } from './create-contentful-api'
 import {
   ChainOptions,
-  ChainOptionWithAllLocalesAndWithLinkResolution,
+  ChainOptionWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks,
   ChainOptionWithAllLocalesAndWithoutLinkResolution,
-  ChainOptionWithLinkResolution,
+  ChainOptionWithAllLocalesAndWithLinkResolutionAndWithoutUnresolvableLinks,
+  ChainOptionWithLinkResolutionAndWithUnresolvableLinks,
   ChainOptionWithoutLinkResolution,
+  ChainOptionWithLinkResolutionAndWithoutUnresolvableLinks,
   DefaultChainOption,
-  isClientWithAllLocalesAndWithLinkResolution,
+  isClientWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks,
   isClientWithAllLocalesAndWithoutLinkResolution,
-  isClientWithLinkResolution,
+  isClientWithAllLocalesAndWithLinkResolutionAndWithoutUnresolvableLinks,
+  isClientWithLinkResolutionAndWithUnresolvableLinks,
   isClientWithoutLinkResolution,
+  isClientWithLinkResolutionAndWithoutUnresolvableLinks,
 } from './utils/client-helpers'
 
-function create<OptionsType, ClientType>(
+function create<OptionsType extends ChainOptions, ClientType>(
   { http, getGlobalOptions }: CreateContentfulApiParams,
   options: OptionsType,
   makeInnerClient: (options: OptionsType) => ClientType
```

Comment:
Don't we need an additional

```typescript
  Object.defineProperty(response, 'withoutUnresolvableLinks', {
    get: () => makeInnerClient({ ...options, ... }),
  })
```
in order to make chains like `client.withAllLocales.withoutUnresolvableLinks.getEntry` work?

Reply:
you're right! fixed